### DISCUSSION TITLE
v0.27.0

### DISCUSSION THREAD

**Discussion by 1715173329** [1715173329/UnblockNeteaseMusic#880] [Announcements]
## TL; DR
- Renewed default certificates
- Fixed compatibility with the latest NCM clients
- Fixed Kuwo, pyNCMd and QQ provider
- Supported follow source order & disable upgrade check
- Improved the documentation
- Other fixes and improvements

## Breaking changes
- [sec!: renew cert](https://github.com/UnblockNeteaseMusic/server/commit/2935d4020b22c7d26ebe2a557f32574524dc1383) #614
  Root CA is changed, you need trust the new cert manually, see [安裝 UNM 的 HTTPS 憑證](https://github.com/UnblockNeteaseMusic/server/discussions/426) for details.

## Fixes
- [fix: Do not colorize if printing to non-TTY device](https://github.com/UnblockNeteaseMusic/server/commit/eac69ed06645a3bb5ca07b472ba01d7e89246570) #798
- [fix(hook): ERR_OSSL_EVP_WRONG_FINAL_BLOCK_LENGTH](https://github.com/UnblockNeteaseMusic/server/commit/f0d4a652b4096c644919d909c57927fb0d9e1e14) #834
- [fix(hook): rework with latest PC clients](https://github.com/UnblockNeteaseMusic/server/commit/10103d53453897ed66fc8ae6abdee922f7e0ddb9) #859
- [fix(hook): add missing encoding tag](https://github.com/UnblockNeteaseMusic/server/commit/9358c8116a871608a8dc85e22bb90cc7744459c8) #840
- [fix(hook): fix working with iOS's search page](https://github.com/UnblockNeteaseMusic/server/commit/501de1f2e3a089a5f74f9704ed6ec8066a573cb4)
- [fix(provider/qq): rework search API](https://github.com/UnblockNeteaseMusic/server/commit/34c0805b41abb10d953a89a4a924e254355dd680) #865
- [fix(webpack): add missing shebang in precompiled bundles](https://github.com/UnblockNeteaseMusic/server/commit/6a36bd3a0122d51a1e6b7d0c369c7e2f44335ab0) #878

## Features
- [feat(npm): add start/dev script](https://github.com/UnblockNeteaseMusic/server/commit/2443fa5ac10f5e879c24e95d92d9f857cc870203)
- [feat(hook): support disable upgrade check](https://github.com/UnblockNeteaseMusic/server/commit/b07709a5b93a14ed1328a71dc269125b1725eb67) #862
- [feat(match): follow source order](https://github.com/UnblockNeteaseMusic/server/commit/ba3374dc058cb068ea8edb226eb5317f6989c64e) #866
- [feat(nw.js): add environment variables support](https://github.com/UnblockNeteaseMusic/server/commit/4ebaefe1326c728b3adaf71fd941043282a76432) #876

## Docs
- [docs(README): update note for youtube and pyncmd](https://github.com/UnblockNeteaseMusic/server/commit/2c71983fd53ff4c84fe14e5813ea942a2f454267)
- [docs: add missing description for port settings](https://github.com/UnblockNeteaseMusic/server/commit/4b0b71809fa7d293c8e806b1247f57aba7d9270e) #817

## Chores
- Update dependencies to the latest

## New Contributors
- @mos9527 made their first contribution in https://github.com/UnblockNeteaseMusic/server/pull/758
- @FrzMtrsprt made their first contribution in https://github.com/UnblockNeteaseMusic/server/pull/798
- @Tiger2lala made their first contribution in https://github.com/UnblockNeteaseMusic/server/pull/817
- @msylgj made their first contribution in https://github.com/UnblockNeteaseMusic/server/pull/840

**Full Changelog**: https://github.com/UnblockNeteaseMusic/server/compare/v0.27.0-rc.6...v0.27.0

<hr /><em>This discussion was created from the release <a href='https://github.com/UnblockNeteaseMusic/server/releases/tag/v0.27.0'>v0.27.0</a>.</em>
